src/utils/checkDesignName.ts

```diff
@@ -0,0 +1,18 @@
+import { DESIGN_NAME_REGULAR_EXPRESSION } from "@/constants/courtData";
+
+const checkDesignName = (name: string, nameList: string[]) => {
+  // when canBeBlank is false, check if name is blank
+  if (name.length === 0) {
```

Comment:
Might need the trim here. I see you have trim when you call checkDesignName else where every time.